lib/smblib/smblib-util.c

```diff
@@ -734,12 +734,12 @@ int SMB_TreeDisconnect(SMB_Tree_Handle Tree_Handle, BOOL discard)
     if (discard == TRUE) { /* Unlink it and free it ... */
 
         if (Tree_Handle -> next == NULL)
-            Tree_Handle -> con -> first_tree = Tree_Handle -> prev;
+            Tree_Handle -> con -> last_tree = Tree_Handle -> prev;
         else
             Tree_Handle -> next -> prev = Tree_Handle -> prev;
 
         if (Tree_Handle -> prev == NULL)
-            Tree_Handle -> con -> last_tree = Tree_Handle -> next;
+            Tree_Handle -> con -> first_tree = Tree_Handle -> next;
```

Comment:
> PR title: Fix tree linking bug of first_tree and last_tree

> PR description: Also fix the reversed SMB_TreeDisconnect unlinking, which hid this bug.

The combination implies that the "linking" bug in PR title had no effect on Squid because that "linking" bug was "hidden" by a bug inside SMB_TreeDisconnect(). Great, we got lucky!

However, it is not clear (to me), what effect, if any, that SMB_TreeDisconnect() "unlinking" bug had on Squid. @MegaManSec, can you tell what that affect was? When answering that question, please assume Squid v7 context (see my other change request for why the answer to this question is probably "none" for master/v8).